pandas/_libs/lib.pyx

```diff
@@ -2343,14 +2367,6 @@ def maybe_convert_objects(ndarray[object] objects, bint try_float=False,
         elif util.is_float_object(val):
             floats[i] = complexes[i] = val
             seen.float_ = True
-        elif util.is_datetime64_object(val):
```

Comment:
unrelated to the rest of the PR, this chunk of code is just redundant with a chunk below